### PR TITLE
Add docs CI and fix some bugs in it

### PR DESCRIPTION
**Pull request by QuLogic**
This copies all the setup from Matplotlib so we get doc builds from Circle CI with posted build artifacts, errors, warnings, etc., on PRs.

This is based on #88.

### CODE REVIEW

**Comment by QuLogic** [comment]
Hmm, bit confused why installing extras with `.[tests]` doesn't work here, but does locally. `pip` is the latest version here AFAICT.

**Comment by ksunden** [comment]
the call has `--no-deps` for some reason, which is why the `[tests]` deps are not installed

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -30,12 +30,11 @@ jobs:
...
       - name: Install Python dependencies
         run: |
-          python -m pip install --upgrade pip setuptools wheel
-          python -m pip install --upgrade pytest pytest-cov pytest-xdist
+          python -m pip install --upgrade pip
 
       - name: Install cycler
         run: |
-          python -m pip install --no-deps .
+          python -m pip install --no-deps .[tests]
```
Comment (ksunden) [inline .github/workflows/tests.yml:37]:
```suggestion
          python -m pip install .[tests]
```

**Comment by QuLogic** [comment]
> the call has --no-deps for some reason, which is why the [tests] deps are not installed

Ah, good catch; I saw that a few times but it didn't register for some reason.

**Review by oscargus — APPROVED** [review]

**Review by ksunden — APPROVED** [review]

**Comment by QuLogic** [comment]
Oh, this seems to have failed because `cycler` doesn't have an `O` variable in the `Makefile`, only `SPHINXOPTS`. I'll open a followup now we have the CI working here.

**Pull request merged**